### PR TITLE
Clean-up windowsdesktop repository

### PR DESCRIPTION
**Pull request by ViktorHofer**
Doing the clean-up as the windowdesktop repository is just a straight copy from core-setup. Meanwhile, a lot changed and infrastructure is vastly improved and simplified. Bringing the improvements from runtime and arcade into this repository.

This is also a preparation step for bringing Microsoft.Windows.Compatibility into windowsdesktop.

- Remove the unneeded buildtools local task publishing infrastructure
- Remoe old infrastructure that was meanwhile replaced by Arcade defaults
- Move pipeline logic into eng/pipelines instead of directly in eng
- Update NoTargets SDK and use Traversal SDK for building the repo
- Remove dotnet7 feeds from NuGet.config
- Remove the daily builds table that wasn't updated anymore since .NET 5.
- Remove Subsets logic which just adds complexity in this repository.
- Remove unused dependencies from Versions.props
- Remove unused files (publish-final.proj, descriptions.json, badge.proj, ...)
- Simplify testing infrastructure and clean-up unused code.
- Remove the sln file that isn't useful in this repository.

cc @ericstj as I mentioned this in our yesterday's 1:1.

### CODE REVIEW

File: Directory.Build.props [hunk 1]
```diff
@@ -2,362 +2,58 @@
...
-      </PropertyGroup>
-    </When>
-    <Otherwise>
-      <PropertyGroup>
-        <ConfigurationErrorMsg>$(ConfigurationErrorMsg);Unknown ConfigurationGroup [$(ConfigurationGroup)] specificed in your project.</ConfigurationErrorMsg>
-      </PropertyGroup>
-    </Otherwise>
-  </Choose>
+    <HostOS>linux</HostOS>
+    <HostOS Condition="$([MSBuild]::IsOSPlatform('OSX'))">osx</HostOS>
```
Comment (mmitche) [inline Directory.Build.props:35]: Needed? There are no non-windows platforms supported.

File: Directory.Build.props [hunk 2]
```diff
@@ -2,362 +2,58 @@
...
+    <HostRuntimeIdentifier Condition="'$(MSBuildRuntimeType)' == 'core'">$([System.Runtime.InteropServices.RuntimeInformation]::RuntimeIdentifier)</HostRuntimeIdentifier>
+    <HostRuntimeIdentifier Condition="'$(MSBuildRuntimeType)' != 'core'">win-$([System.Runtime.InteropServices.RuntimeInformation]::OSArchitecture.ToString().ToLowerInvariant)</HostRuntimeIdentifier>
 
-  <PropertyGroup Condition="'$(RuntimeIdentifier)' == '' and '$(HostRuntimeIdentifier)' != ''">
-    <RuntimeIdentifier>$(HostRuntimeIdentifier.Remove($(HostRuntimeIdentifier.LastIndexOf('-'))))-$(TargetArchitecture)</RuntimeIdentifier>
+    <RuntimeIdentifier Condition="'$(RuntimeIdentifier)' == ''">$(HostRuntimeIdentifier.Remove($(HostRuntimeIdentifier.LastIndexOf('-'))))-$(TargetArchitecture)</RuntimeIdentifier>
   </PropertyGroup>
 
-  <!-- Portable -->
   <PropertyGroup Condition="'$(PortableBuild)' == 'true'">
```
Comment (mmitche) [inline Directory.Build.props:48]: Is portable vs. non a meaningful distinction here?

File: README.md [hunk 3]
```diff
@@ -5,7 +5,7 @@ supported platforms.
 
 ## Getting started
 
-* [.NET Core 3.1 SDK](https://dotnet.microsoft.com/download/dotnet-core)
+* [.NET 7 SDK](https://dotnet.microsoft.com/en-us/download/dotnet)
```
Comment (mmitche) [inline README.md:8]: I think you need the .NET 8 SDK now.

File: azure-pipelines.yml [hunk 4]
```diff
@@ -88,6 +86,29 @@ stages:
...
+  - template: /eng/common/templates/post-build/post-build.yml
     parameters:
       publishingInfraVersion: 3
+      validateDependsOn:
+      - PrepareForPublish
+      enableSymbolValidation: false
+      enableSigningValidation: false
+      enableNugetValidation: false
+      enableSourceLinkValidation: false
+      publishInstallersAndChecksums: true
```
Comment (mmitche) [inline azure-pipelines.yml:99]: This is the default now, no need to include.

File: src/publish/Directory.Build.props [hunk 5]
```diff
@@ -0,0 +1,10 @@
+<Project>
+
+  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory).., Directory.Build.props))\Directory.Build.props" />
+
+  <PropertyGroup>
+    <GenerateChecksumsTaskDir>$(ArtifactsBinDir)GenerateChecksums\</GenerateChecksumsTaskDir>
```
Comment (mmitche) [inline src/publish/Directory.Build.props:6]: Arcade now has a task for this. Can we use that instead?

File: src/publish/GenerateChecksums/GenerateChecksums.csproj [hunk 6]
```diff
@@ -0,0 +1,15 @@
+<Project Sdk="Microsoft.NET.Sdk">
```
Comment (mmitche) [inline src/publish/GenerateChecksums/GenerateChecksums.csproj:1]: I don't think this is actually necessary any longer.

File: src/windowsdesktop/tests/Microsoft.WindowsDesktop.App.Tests/Microsoft.WindowsDesktop.App.Tests.csproj [hunk 7]
```diff
@@ -0,0 +1,16 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <ItemGroup>
+    <!-- Ensure the packaging projects build first. Don't do this in VS: the build takes too long and isn't incremental. -->
+    <ProjectReference Include="..\..\src\sfx\**\*.sfxproj"
+                      ReferenceOutputAssembly="false"
+                      Condition="'$(SkipBuildingSharedFrameworkTestDependencies)' != 'true' and
+                                 '$(BuildingInsideVisualStudio)' != 'true'" />
+    <ProjectReference Include="..\TestUtils\TestUtils.csproj" />
```
Comment (mmitche) [inline src/windowsdesktop/tests/Microsoft.WindowsDesktop.App.Tests/Microsoft.WindowsDesktop.App.Tests.csproj:9]: What's the reason for the reference to the empty project here?

**Comment by ViktorHofer** [comment]
@mmitche thanks for the great feedback. Addressed it in commit 2-5.

New official build for validation: https://dnceng.visualstudio.com/internal/_build/results?buildId=2142219&view=results

**Comment by ViktorHofer** [comment]
cc @dreddy-work and @merriemcgaw just FYI. This PR removed a ton of obsolete code and makes working in windowsdesktop easier.

File: Subsets.props [hunk 8]
```diff
@@ -1,90 +0,0 @@
-<Project>
```
Comment (carlossanlop) [inline Subsets.props:1]: Should this file be deleted? Or why was it preserved?

**Pull request merged**